programs/mango-v4/src/health/account_retriever.rs

```diff
@@ -225,12 +258,13 @@ impl<'a, 'info> ScannedBanksAndOracles<'a, 'info> {
     ) -> Result<(&mut Bank, I80F48, Option<(&mut Bank, I80F48)>)> {
         if token_index1 == token_index2 {
             let index = self.bank_index(token_index1)?;
+            let price = {
+                let bank = self.banks[index].load_fully_unchecked::<Bank>()?;
+                let oracle_acc_infos = self.create_oracle_infos(index, &bank.fallback_oracle);
```

Comment:
this function takes the oracle_index as the first argument, not the bank index, so this is likely a bug. It might be helpful to have a `bank_oracle_infos()` fn where you pass in the bank index, it reads that to get the fallback, computes the oracle index etc?

Reply:
For a ScanningAccountRetriever, bank index == oracle index (since banks and oracles are stored in separate vectors)

This is not the case in a FixedAccountRetriever however and `create_oracle_infos` uses `self.ais[index]` to pick the oralce account  vs `self.oracles[index]` (since all account infos are stored together)

Reply:
Right, got it, not an issue.